### PR TITLE
Support tabcompleting commands

### PR DESCRIPTION
**Pull request by mia-pi-git**
Self-explanatory. Will merge in a day or so if no one has any complaints, so please eyeball sooner rather than later.

### CODE REVIEW

File: play.pokemonshowdown.com/js/client-chat.js [hunk 1]
```diff
@@ -357,43 +357,73 @@
...
+					return x.startsWith(prefix);
+				}).length;
+				var isCommandSearch = text.startsWith('/') || text.startsWith('!');
+				if (isCommandSearch && shouldSearchCommands) {
+					if (this.tabComplete.searchPending) return true; // wait
+					this.tabComplete.isCommand = true;
+					this.tabComplete.searchPending = true;
+					app.send('/crq cmdsearch ' + text);
+					var self = this;
+					app.once('response:cmdsearch', function (data) {
```
Comment (Karthik99999) [inline play.pokemonshowdown.com/js/client-chat.js:371]: Isn't the handler supposed to be defined before the `/crq` is sent?

Comment (mia-pi-git) [reply play.pokemonshowdown.com/js/client-chat.js:371]: It works either way. But I suppose there's no real harm in being cautious.

**Comment by DaWoblefet** [comment]
Typing /z (or anything else that doesn't return any commands) sends the request twice.

**Comment by mia-pi-git** [comment]
> Typing /z (or anything else that doesn't return any commands) sends the request twice.

This has been fixed.

**Pull request merged**